### PR TITLE
Workaround for @parcel/transformer-typescript-types in monorepo

### PR DESCRIPTION
**Pull request by mischnic**
# ↪️ Pull Request

Closes https://github.com/parcel-bundler/parcel/issues/3891

Workaround for https://github.com/microsoft/TypeScript/issues/39547

This works around a (likely) TS bug by:
1. if a package.json file references a `types` file that doesn't exist, rewrite it to the `source` field's content (but with `.d.ts` instead of `.ts` because TS apparently requires that).
2. This redirection list is then used in `fileExists` and `getSourceFile` to return the file `src/index.ts` when the virtual `src/index.d.ts` is requested.

@devongovett is this what you had in mind?

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 5.93s | -738.00ms 🚀 |
| Cached | 2.16s | -143.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 305.00ms | +77.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 306.00ms | +79.00ms ⚠️ |
| dist/legacy/index.42516b4a.js | 1.10kb | +0.00b | 955.00ms | -808.00ms 🚀 |
| dist/modern/index.8a1ced31.js | 1.10kb | +0.00b | 610.00ms | -1.15s 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 858.00ms | +185.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 855.00ms | +184.00ms ⚠️ |
| dist/legacy/index.0e158b6b.css | 77.00b | +0.00b | 1.35s | -417.00ms 🚀 |
| dist/modern/index.e480862d.css | 77.00b | +0.00b | 1.35s | -72.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 209.00ms | -67.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 209.00ms | -67.00ms 🚀 |
| dist/legacy/index.42516b4a.js | 1.10kb | +0.00b | 360.00ms | -120.00ms 🚀 |
| dist/modern/index.8a1ced31.js | 1.10kb | +0.00b | 359.00ms | -120.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 362.00ms | -120.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 359.00ms | -120.00ms 🚀 |
| dist/legacy/index.0e158b6b.css | 77.00b | +0.00b | 360.00ms | -120.00ms 🚀 |
| dist/modern/index.e480862d.css | 77.00b | +0.00b | 359.00ms | -120.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 27.34s | +1.45s ⚠️ |
| Cached | 2.38s | -32.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 158.00ms | -37.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.42kb | +0.00b | 232.00ms | -72.00ms 🚀 |
| dist/PermalinkedComment.c85d137d.js | 4.22kb | +0.00b | 230.00ms | -47.00ms 🚀 |
| dist/UserProfile.30e15baf.js | 1.62kb | +0.00b | 229.00ms | -48.00ms 🚀 |
| dist/NotFound.5aa6b041.js | 387.00b | +0.00b | 229.00ms | -48.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 168.00ms | -54.00ms 🚀 |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.86m | -3.22s |
| Cached | 5.21s | +54.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/pdfRenderer.cd742f5e.js | 1.11mb | +0.00b | 52.03s | -3.82s 🚀 |
| dist/editorView.ae59b49f.js | 585.98kb | +0.00b | 1.33m | -5.67s 🚀 |
| dist/popup.4a66e15e.js | 170.62kb | +0.00b | 52.03s | +17.23s ⚠️ |
| dist/EmojiPickerComponent.bab709db.js | 97.89kb | +0.00b | 10.67s | -1.71s 🚀 |
| dist/Toolbar.33a83ae3.js | 92.34kb | +0.00b | 55.62s | -3.03s 🚀 |
| dist/media-viewer.1d9322e7.js | 69.09kb | +0.00b | 14.81s | -2.01s 🚀 |
| dist/card.ce67dc35.js | 55.09kb | +0.00b | 13.52s | -1.24s 🚀 |
| dist/card.1007a95f.js | 52.86kb | +0.00b | 53.50s | -4.05s 🚀 |
| dist/Modal.e3bd77d0.js | 41.40kb | +0.00b | 4.90s | -1.41s 🚀 |
| dist/component.59edff1f.js | 30.78kb | +0.00b | 3.03s | +384.00ms ⚠️ |
| dist/component.37867f50.js | 22.51kb | +0.00b | 5.77s | -550.00ms 🚀 |
| dist/DatePicker.cbe44a9b.js | 21.17kb | +0.00b | 10.17s | -587.00ms 🚀 |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 10.17s | -587.00ms 🚀 |
| dist/js.48be70a4.js | 16.40kb | +0.00b | 4.90s | -1.42s 🚀 |
| dist/dropzone.c8c03b6a.js | 14.36kb | +0.00b | 52.95s | -4.16s 🚀 |
| dist/ui.cee430a3.js | 13.97kb | +0.00b | 58.88s | +44.21s ⚠️ |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 54.67s | -46.85s 🚀 |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 12.17s | -1.62s 🚀 |
| dist/component.11476a92.js | 6.05kb | +0.00b | 5.77s | -550.00ms 🚀 |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 1.69m | +1.43m ⚠️ |
| dist/media-viewer.bc661bce.js | 3.92kb | +0.00b | 14.77s | -2.05s 🚀 |
| dist/EmojiPickerComponent.624bf3f4.js | 3.58kb | +0.00b | 11.64s | +884.00ms ⚠️ |
| dist/index.393afe59.css | 3.46kb | +0.00b | 5.16s | -54.47s 🚀 |
| dist/png-chunks-extract.8d15a914.js | 3.41kb | +0.00b | 4.90s | -1.41s 🚀 |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 29.47s | -2.84s 🚀 |
| dist/Modal.2835744c.js | 3.08kb | +0.00b | 4.90s | +646.00ms ⚠️ |
| dist/clipboard.0244488f.js | 2.75kb | +0.00b | 52.77s | -3.35s 🚀 |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 4.90s | +2.26s ⚠️ |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 15.04s | -2.02s 🚀 |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 6.69s | +2.44s ⚠️ |
| dist/date.d49e64f8.js | 1.73kb | +0.00b | 6.28s | -955.00ms 🚀 |
| dist/16.f40c1329.js | 1.73kb | +0.00b | 8.14s | +1.10s ⚠️ |
| dist/images.80270b1f.js | 1.67kb | +0.00b | 6.36s | -49.30s 🚀 |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 5.77s | -1.44s 🚀 |
| dist/feedback.8b7c7928.js | 1.64kb | +0.00b | 8.57s | +917.00ms ⚠️ |
| dist/browser.16c438d4.js | 1.61kb | +0.00b | 52.77s | -3.37s 🚀 |
| dist/workerHasher.f66f00c7.js | 1.54kb | +0.00b | 54.55s | -46.97s 🚀 |
| dist/list-number.4632b459.js | 1.46kb | +0.00b | 6.46s | -900.00ms 🚀 |
| dist/status.cccf04be.js | 1.46kb | +0.00b | 6.54s | -993.00ms 🚀 |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 6.03s | -677.00ms 🚀 |
| dist/code.af9ca2b3.js | 1.38kb | +0.00b | 6.28s | -955.00ms 🚀 |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 5.77s | -1.44s 🚀 |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 6.03s | -677.00ms 🚀 |
| dist/16.a4b392c2.js | 1.32kb | +0.00b | 3.02s | +379.00ms ⚠️ |
| dist/16.249c644a.js | 1.31kb | +0.00b | 6.17s | -1.07s 🚀 |
| dist/link.d12cc130.js | 1.30kb | +0.00b | 6.36s | -1.00s 🚀 |
| dist/heading6.4c4017db.js | 1.30kb | +0.00b | 8.57s | +917.00ms ⚠️ |
| dist/heading3.93606b9c.js | 1.29kb | +0.00b | 6.65s | -1.01s 🚀 |
| dist/16.ab5c0088.js | 1.27kb | +0.00b | 6.17s | -873.00ms 🚀 |
| dist/emoji.e603e70e.js | 1.23kb | +0.00b | 6.36s | -49.30s 🚀 |
| dist/16.afee1494.js | 1.23kb | +0.00b | 6.17s | -872.00ms 🚀 |
| dist/16.e60b4d77.js | 1.21kb | +0.00b | 3.02s | +379.00ms ⚠️ |
| dist/16.7c7a40c9.js | 1.19kb | +0.00b | 6.17s | -872.00ms 🚀 |
| dist/16.06d5a67b.js | 1.18kb | +0.00b | 6.28s | -954.00ms 🚀 |
| dist/heading5.dde15cec.js | 1.17kb | +0.00b | 8.57s | +917.00ms ⚠️ |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 8.57s | -2.18s 🚀 |
| dist/media-card-analytics-error-boundary.6a25b970.js | 1.15kb | +0.00b | 29.46s | -2.83s 🚀 |
| dist/16.2096b9f3.js | 1.14kb | +0.00b | 3.02s | +380.00ms ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 6.03s | -677.00ms 🚀 |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 7.71s | +503.00ms ⚠️ |
| dist/16.1b342686.js | 1.13kb | +0.00b | 6.03s | -677.00ms 🚀 |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 6.03s | -677.00ms 🚀 |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 5.77s | -1.44s 🚀 |
| dist/heading2.e219309a.js | 1.11kb | +0.00b | 6.65s | -885.00ms 🚀 |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 3.02s | -1.23s 🚀 |
| dist/component.40194190.js | 1.07kb | +0.00b | 6.67s | +2.42s ⚠️ |
| dist/mention.b1828447.js | 1.07kb | +0.00b | 6.46s | -899.00ms 🚀 |
| dist/heading4.5316f6f2.js | 1.06kb | +0.00b | 6.67s | -990.00ms 🚀 |
| dist/16.317b9164.js | 1.05kb | +0.00b | 6.17s | -872.00ms 🚀 |
| dist/layout.ac6734f9.js | 1.05kb | +0.00b | 6.36s | -1.00s 🚀 |
| dist/divider.d2be55a1.js | 1.03kb | +0.00b | 6.36s | -49.30s 🚀 |
| dist/quote.f4e09143.js | 1.02kb | +0.00b | 6.54s | -993.00ms 🚀 |
| dist/action.cb287eb0.js | 1.00kb | +0.00b | 6.28s | -955.00ms 🚀 |
| dist/decision.91f3af14.js | 1010.00b | +0.00b | 6.28s | -49.38s 🚀 |
| dist/panel-warning.68b04096.js | 1008.00b | +0.00b | 6.54s | -910.00ms 🚀 |
| dist/list.057cb664.js | 978.00b | +0.00b | 6.46s | -900.00ms 🚀 |
| dist/heading1.fa66b375.js | 976.00b | +0.00b | 6.65s | -885.00ms 🚀 |
| dist/panel-error.c4b05f6f.js | 906.00b | +0.00b | 6.46s | -994.00ms 🚀 |
| dist/panel.9dbc048e.js | 897.00b | +0.00b | 6.54s | -910.00ms 🚀 |
| dist/table.b35545be.js | 889.00b | +0.00b | 6.65s | -885.00ms 🚀 |
| dist/panel-success.7eb0809a.js | 845.00b | +0.00b | 6.54s | -910.00ms 🚀 |
| dist/panel-note.ec8386b7.js | 841.00b | +0.00b | 6.46s | -994.00ms 🚀 |
| dist/media-picker-analytics-error-boundary.553055f6.js | 837.00b | +0.00b | 51.79s | +17.18s ⚠️ |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 58.90s | +44.22s ⚠️ |
| dist/media-card-analytics-error-boundary.57503222.js | 833.00b | +0.00b | 53.50s | -48.02s 🚀 |
| dist/media-viewer-analytics-error-boundary.bbe40a02.js | 757.00b | +0.00b | 14.77s | -2.05s 🚀 |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 12.11s | -620.00ms 🚀 |
| dist/simpleHasher.f8e69464.js | 528.00b | +0.00b | 58.88s | +44.21s ⚠️ |
| dist/simpleHasher.32d9352e.js | 528.00b | +0.00b | 54.55s | -46.96s 🚀 |
| dist/index.html | 119.00b | +0.00b | 5.77s | +4.91s ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.3417749f.js | 2.73mb | +0.00b | 684.00ms | +147.00ms ⚠️ |
| dist/pdfRenderer.cd742f5e.js | 1.11mb | +0.00b | 681.00ms | +152.00ms ⚠️ |
| dist/editorView.ae59b49f.js | 585.98kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/popup.4a66e15e.js | 170.62kb | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/EmojiPickerComponent.bab709db.js | 97.89kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/Toolbar.33a83ae3.js | 92.34kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/media-viewer.1d9322e7.js | 69.09kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/card.ce67dc35.js | 55.09kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/card.1007a95f.js | 52.86kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/Modal.e3bd77d0.js | 41.40kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/component.59edff1f.js | 30.78kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/component.37867f50.js | 22.51kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/DatePicker.cbe44a9b.js | 21.17kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/js.48be70a4.js | 16.40kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/dropzone.c8c03b6a.js | 14.36kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/ui.cee430a3.js | 13.97kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.9af6e5fc.js | 8.67kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/component.11476a92.js | 6.05kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/media-viewer.bc661bce.js | 3.92kb | +0.00b | 681.00ms | +153.00ms ⚠️ |
| dist/EmojiPickerComponent.624bf3f4.js | 3.58kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/index.393afe59.css | 3.46kb | +0.00b | 668.00ms | +139.00ms ⚠️ |
| dist/png-chunks-extract.8d15a914.js | 3.41kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/Modal.2835744c.js | 3.08kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/clipboard.0244488f.js | 2.75kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/EmojiUploadComponent.88a444df.js | 2.18kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 684.00ms | +147.00ms ⚠️ |
| dist/date.d49e64f8.js | 1.73kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/16.f40c1329.js | 1.73kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/images.80270b1f.js | 1.67kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/feedback.8b7c7928.js | 1.64kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/browser.16c438d4.js | 1.61kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/workerHasher.9bb425b1.js | 1.54kb | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/workerHasher.f66f00c7.js | 1.54kb | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/list-number.4632b459.js | 1.46kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/status.cccf04be.js | 1.46kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/code.af9ca2b3.js | 1.38kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.a4b392c2.js | 1.32kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/16.249c644a.js | 1.31kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/link.d12cc130.js | 1.30kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/heading6.4c4017db.js | 1.30kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/heading3.93606b9c.js | 1.29kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/16.ab5c0088.js | 1.27kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/emoji.e603e70e.js | 1.23kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.afee1494.js | 1.23kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.e60b4d77.js | 1.21kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/16.7c7a40c9.js | 1.19kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/16.06d5a67b.js | 1.18kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/heading5.dde15cec.js | 1.17kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.6a25b970.js | 1.15kb | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/16.2096b9f3.js | 1.14kb | +0.00b | 683.00ms | +146.00ms ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.1b342686.js | 1.13kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/heading2.e219309a.js | 1.11kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 684.00ms | +147.00ms ⚠️ |
| dist/component.40194190.js | 1.07kb | +0.00b | 684.00ms | +147.00ms ⚠️ |
| dist/mention.b1828447.js | 1.07kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/heading4.5316f6f2.js | 1.06kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/16.317b9164.js | 1.05kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/layout.ac6734f9.js | 1.05kb | +0.00b | 683.00ms | +147.00ms ⚠️ |
| dist/divider.d2be55a1.js | 1.03kb | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/quote.f4e09143.js | 1.02kb | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/action.cb287eb0.js | 1.00kb | +0.00b | 682.00ms | +146.00ms ⚠️ |
| dist/decision.91f3af14.js | 1010.00b | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/panel-warning.68b04096.js | 1008.00b | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/list.057cb664.js | 978.00b | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/heading1.fa66b375.js | 976.00b | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/panel-error.c4b05f6f.js | 906.00b | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/panel.9dbc048e.js | 897.00b | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/table.b35545be.js | 889.00b | +0.00b | 681.00ms | +146.00ms ⚠️ |
| dist/panel-success.7eb0809a.js | 845.00b | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/panel-note.ec8386b7.js | 841.00b | +0.00b | 682.00ms | +147.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.553055f6.js | 837.00b | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 681.00ms | +147.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.57503222.js | 833.00b | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.bbe40a02.js | 757.00b | +0.00b | 669.00ms | +141.00ms ⚠️ |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/simpleHasher.f8e69464.js | 528.00b | +0.00b | 682.00ms | +148.00ms ⚠️ |
| dist/simpleHasher.32d9352e.js | 528.00b | +0.00b | 668.00ms | +140.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 685.00ms | +147.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/270928470435955212)**

**Comment by devongovett** [comment]
> @devongovett is this what you had in mind?

Yeah basically. That's cool that it works! 😲

File: packages/utils/ts-utils/src/CompilerHost.js [hunk 1]
```diff
@@ -1,22 +1,75 @@
...
+        json.types != null &&
+        json.source != null &&
+        !super.fileExists(
+          path.posix.join(path.posix.dirname(filePath), json.types),
+        )
+      ) {
+        let source = path.posix.join(path.posix.dirname(filePath), json.source);
+        let fakeTypes =
+          source.slice(0, -path.posix.extname(source).length) + '.d.ts';
+        this.redirectTypes.set(fakeTypes, source);
```
Comment (devongovett) [inline packages/utils/ts-utils/src/CompilerHost.js:38]: Lol, so TS requires it to be a `.d.ts` file but doesn't validate that when parsing? 🤦

Comment (mischnic) [reply packages/utils/ts-utils/src/CompilerHost.js:38]: Something along those lines, I don't quite remember.

**Pull request merged**